tests/ops/qubit/test_non_parametric_ops.py

```diff
@@ -918,16 +918,37 @@ def test_pauliz_general_power(self, n):
         assert op_pow[0].__class__ is qml.PhaseShift
         assert qml.math.allclose(op_pow[0].data[0], np.pi * (n % 2))
 
-    @pytest.mark.parametrize("n", (0.5, 2.5, -1.5))
-    def test_ISWAP_sqaure_root(self, n):
-        """Test that SISWAP is the square root of ISWAP."""
-        op = qml.ISWAP(wires=(0, 1))
-
-        assert op.pow(n)[0].__class__ is qml.SISWAP
+    @pytest.mark.parametrize(
+        "n, expected",
+        [
+            (0, []),
+            (4, []),
+            (-4, []),
+            (-3, [qml.ISWAP(wires=(0, 1))]),
+            (5, [qml.ISWAP(wires=(0, 1))]),
+            (0.5, [qml.SISWAP(wires=(0, 1))]),
+            (4.5, [qml.SISWAP(wires=(0, 1))]),
+            (2, [qml.Z(0), qml.Z(1)]),
+            (-2, [qml.Z(0), qml.Z(1)]),
+            (6, [qml.Z(0), qml.Z(1)]),
+        ],
+    )
+    def test_ISWAP_powers(self, n, expected):
+        """Check that the special powers of ISWAP are correct."""
 
-        sqrt_mat = qml.matrix(op.pow, wire_order=[0, 1])(n)
-        sqrt_mat_squared = qml.math.linalg.matrix_power(sqrt_mat, 2)
-        assert qml.math.allclose(sqrt_mat_squared, qml.matrix(op))
+        op = qml.ISWAP(wires=(0, 1))
+        op_mat = qml.matrix(op)
+        pow_ops = op.pow(n)
+        assert pow_ops == expected
+        if not pow_ops:
+            pow_ops.append(qml.I(wires=(0, 1)))
+        mat = qml.matrix(qml.prod(*pow_ops), wire_order=[0, 1])
+        expected = (
+            qml.math.linalg.matrix_power(op_mat, n)
+            if isinstance(n, int)
+            else sp.linalg.fractional_matrix_power(op_mat, n)
+        )
+        assert qml.math.allclose(mat, expected)
```

Comment:
Can we separate the numerical checks against scipy into another test, and then range the params to cover all `np.arange(0, 4, 0.5)`? In this way we can safeguard that our pow is numerically correct in all places we claim to support.

Reply:
So an operator overrides `Operator.pow` only for special handlings of special powers. The general power decomposition is just to repeat the operator `n` number of times if `n` is an integer. We don't really claim to support arbitrary powers.

Reply:
@astralcai I see. So this `Operator.pow` is only for decomposition?

Reply:
I believe so.

Reply:
OK thanks for clarification! If so then it makes sense to me. Just note that previously we could use `qml.ISWAP([0,1]).pow(3)` but now it's blocked.

Reply:
I mean what we get previously was wrong.